dspace/config/crosswalks/oai/metadataFormats/oai_openaire.xsl

```diff
@@ -0,0 +1,1484 @@
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:value-of select="false()"/>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <!-- will try to resolve the field type based on the value -->
+    <xsl:template name="resolveFieldType">
+        <xsl:param name="field"/>
+        <!-- regexp not supported on XSLTv1 -->
+        <xsl:variable name="isHandle">
+            <xsl:call-template name="isHandle">
+                <xsl:with-param name="field" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:variable name="isDOI">
+            <xsl:call-template name="isDOI">
+                <xsl:with-param name="field" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:variable name="isURL">
+            <xsl:call-template name="isURL">
+                <xsl:with-param name="field" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:choose>
+            <xsl:when test="$isHandle = 'true'">
+                <xsl:text>Handle</xsl:text>
+            </xsl:when>
+            <xsl:when test="$isDOI = 'true'">
+                <xsl:text>DOI</xsl:text>
+            </xsl:when>
+            <xsl:when test="$isURL = 'true' and $isHandle = 'false'">
+                <xsl:text>URL</xsl:text>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:text>N/A</xsl:text>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <xsl:template name="resolveResourceTypeGeneral">
+        <xsl:param name="field"/>
+        <xsl:variable name="lc_dc_type">
+            <xsl:call-template name="lowercase">
+                <xsl:with-param name="value" select="$field"/>
+            </xsl:call-template>
+        </xsl:variable>
+        <xsl:choose>
+            <xsl:when test="$lc_dc_type = 'article'">
+                <xsl:text>literature</xsl:text>
+            </xsl:when>
+            <xsl:when test="$lc_dc_type = 'journal article'">
+                <xsl:text>literature</xsl:text>
+            </xsl:when>
+            <xsl:when test="$lc_dc_type = 'book'">
+                <xsl:text>literature</xsl:text>
+            </xsl:when>
+            <xsl:when test="$lc_dc_type = 'book part'">
+                <xsl:text>literature</xsl:text>
+            </xsl:when>
+            <xsl:when test="$lc_dc_type = 'book review'">
+                <xsl:text>literature</xsl:text>
+            </xsl:when>
+            <xsl:when test="$lc_dc_type = 'dataset'">
+                <xsl:text>dataset</xsl:text>
+            </xsl:when>
+            <xsl:when test="$lc_dc_type = 'software'">
+                <xsl:text>software</xsl:text>
+            </xsl:when>
+            <xsl:otherwise>
+                other research product
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+
+    <xsl:template name="resolveResourceTypeURI">
```

Comment:
Please describe this template with a comment.

Reply:
Just add one description.
https://github.com/DSpace/DSpace/pull/2592/commits/7814a9a1df632f03d9d97191637e722078bba317